cubic-server/WorldGroup.cpp

```diff
@@ -4,12 +4,15 @@
 #include "Server.hpp"
 #include "SoundSystem.hpp"
 #include "World.hpp"
+#include "Dimension.hpp"
 #include "logging/Logger.hpp"
+#include "scoreboard/Scoreboard.hpp"
 
 WorldGroup::WorldGroup(std::shared_ptr<Chat> chat):
     _chat(std::move(chat)),
     _soundSystem(new SoundSystem(*this)),
-    _running(false)
+    _running(false),
+    _scoreboard(Server::getInstance()->getScoreboardSystem(), *this)
```

Comment:
maybe there you want to use shared_from_this() and get a shared_ptr instead of a reference